spec.bs

```diff
@@ -223,6 +226,46 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
   </dd>
 </dl>
 
+<div algorithm>
+  The <dfn constructor for=URLPattern lt="URLPattern(input, baseURL)">new URLPattern(|input|, |baseURL|)</dfn> constructor steps are:
+
+  1. Let |init| be null.
+  1. If |input| is a [=scalar value string=] then:
+    1. Set |init| to the result of running [=parse a constructor string=] given |input|.
+    1. Set |init|["{{URLPatternInit/baseURL}}"] to |baseURL|.
+  1. Else:
+    1. [=Assert=]: |input| is a {{URLPatternInit}}.
+    1. If |baseURL| is given, then throw a {{TypeError}}.
+    1. Set |init| to |input|.
+  1. Let |protocol| be null.
+  1. Let |username| be null.
+  1. Let |password| be null.
+  1. Let |hostname| be null.
+  1. Let |port| be null.
+  1. Let |pathname| be null.
+  1. Let |search| be null.
+  1. Let |hash| be null.
+  1. Let |processedInit| be the result of [=process a URLPatternInit=] given |init|, "`pattern`", |protocol|, |username|, |password|, |hostname|, |port|, |pathname|, |search|, and |hash|.
+  1. Set |protocol| to |processedInit|["{{URLPatternInit/protocol}}"].
+  1. Set |username| to |processedInit|["{{URLPatternInit/username}}"].
+  1. Set |password| to |processedInit|["{{URLPatternInit/password}}"].
+  1. Set |hostname| to |processedInit|["{{URLPatternInit/hostname}}"].
+  1. Set |port| to |processedInit|["{{URLPatternInit/port}}"].
+  1. Set |pathname| to |processedInit|["{{URLPatternInit/pathname}}"].
+  1. Set |search| to |processedInit|["{{URLPatternInit/search}}"].
+  1. Set |hash| to |processedInit|["{{URLPatternInit/hash}}"].
+  1. If |protocol| is a [=special scheme=] and |port| is its corresponding [=url/port=], then set |port| to the empty string.
```

Comment:
It looks like there's a definition for this: https://url.spec.whatwg.org/#default-port